src/util.cpp

```diff
@@ -1120,6 +1126,14 @@ std::string toString(const Decl *decl)
 {
     // Reference: https://clang.llvm.org/doxygen/classclang_1_1Decl.html
 
+    if (decl == nullptr)
+    {
+        std::cerr << "Warning: nullptr is not allowed to get the type name of "
+                     "clang::Decl*"
+                  << std::endl;
+        return "Error: [nullptr]";
```

Comment:
This is being used _in_ other template strings, we really want to put a delimiter around this so we can see this clearly in the templates where it would show up.

```suggestion
        return "[nullptr]";
```

```suggestion
        return "std::nullptr_t";
```
I don't see a point in putting "error" in the string itself, since we are already printing the warning.  Putting a verbose error string into the rendered binding is just going to cause even more trouble.

Is there a compiler preprocessor directive or static_assert we could return that would do something meaningful when compiled?

Reply:
> This is being used in other template strings

Hm, I'm not able to find a place this function is used in the core implementation. Could you point me where it is?

Reply:
Hmm, I thought it was for the templates, but maybe it is only for type lookups or error messages.

Reply:
Actually, now that you mention it, what is the case where we hit this?

Should it just be a `std::runtime_error`?  It seems like we would only get `nullptr` in an actual bug situation.

Reply:
We have a dedicated function to get the type name, and these functions are created just for debugging for the case that the dedicated function doesn't work for some reason. I documented this in the header. Should we move these functions to a different namespace or a nested namespace (e.g., `chimera::util::debug`) to be more clear that this is only for debugging?

Reply:
> Actually, now that you mention it, what is the case where we hit this?

Well, this could hit when I pass `nullptr` in debugging? :thinking: 

> Should it just be a `std::runtime_error`? It seems like we would only get `nullptr` in an actual bug situation. We cannot get it spuriously due to bad input, right?

Not sure, if we really need to consider how to deal with `nullptr` because this function is only temporarily used during debugging, and we never commit a code that uses this function.